lib/Model/Iteration.php

```diff
@@ -137,38 +115,44 @@ public function incrementRejectionCount()
      */
     public function getRejectionCount()
     {
-        return $this->rejectionCount;
+        return $this->results->hasResult(ComputedResult::class) ? $this->results->getResult(ComputedResult::class)->getRejectCount() : 0;
     }
 
     /**
      * Return the time taken (in microseconds) to perform this iteration (or
      * NULL if not yet performed.
      *
+     * TODO: Remove this method
+     *
      * @return int
      */
     public function getTime()
     {
-        return $this->time;
+        return $this->results->getResult(TimeResult::class)->getTime();
     }
 
     /**
-     * Return the memory (in bytes) taken to perform this iteration (or
-     * NULL if not yet performed.
+     * Return the revolution time.
      *
-     * @return int
+     * TODO: Remove this method.
+     *
+     * @return float
      */
-    public function getMemory()
+    public function getRevTime()
     {
-        return $this->memory;
+        return $this->getTime() / $this->getVariant()->getRevolutions();
     }
 
     /**
-     * Return the revolution time.
+     * Return the memory (in bytes) taken to perform this iteration (or
+     * NULL if not yet performed.
      *
-     * @return float
+     * TODO: Remove this method.
+     *
+     * @return int
      */
-    public function getRevTime()
+    public function getMemory()
     {
-        return $this->time / $this->getVariant()->getRevolutions();
+        return $this->results->hasResult(MemoryResult::class) ? $this->results->getResult(MemoryResult::class)->getMemory() : null;
```

Comment:
This should be per-revolution, also should pass nb. revolutions to the time (or calculate the rev-time initially).